File: app/src/main/java/com/originb/inkwisenote/modules/smartnotes/SmartNotebookEventListener.java
package com.originb.inkwisenote.modules.smartnotes;

import com.originb.inkwisenote.modules.backgroundjobs.BackgroundOps;
import com.originb.inkwisenote.modules.backgroundjobs.Events;
import com.originb.inkwisenote.modules.repositories.Repositories;
import com.originb.inkwisenote.modules.repositories.SmartNotebookRepository;
import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

public class SmartNotebookEventListener {
    private SmartNotebookRepository smartNotebookRepository;

    public SmartNotebookEventListener() {
        smartNotebookRepository = Repositories.getInstance().getSmartNotebookRepository();
        EventBus.getDefault().register(this);
    }

    @Subscribe(threadMode = ThreadMode.BACKGROUND)
    public void onNotebookDelete(Events.NotebookDeleted notebookToDelete) {
        BackgroundOps.execute(() -> BackgroundOps.execute(() ->
                smartNotebookRepository.deleteSmartNotebook(notebookToDelete.smartNotebook)
        ));
    }

    @Subscribe(threadMode = ThreadMode.BACKGROUND)
    public void onNoteDelete(Events.NoteDeleted noteDeleted) {
        BackgroundOps.execute(() -> BackgroundOps.execute(() ->
                smartNotebookRepository.deleteNoteFromBook(noteDeleted.smartNotebook, noteDeleted.atomicNote)
        ));
    }

}

File: androidApp/src/main/java/com/originb/inkwisenote2/modules/smartnotes/SmartNotebookEventListener.kt
package com.originb.inkwisenote2.modules.smartnotes

import com.originb.inkwisenote2.modules.backgroundjobs.BackgroundOps.Companion.execute
import com.originb.inkwisenote2.modules.backgroundjobs.Events.DeleteNoteCommand
import com.originb.inkwisenote2.modules.backgroundjobs.Events.DeleteNotebookCommand
import com.originb.inkwisenote2.modules.repositories.SmartNotebookRepository
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class SmartNotebookEventListener(private val smartNotebookRepository: SmartNotebookRepository) {
    init {
        // Inject SmartNotebookRepository via Koin
        EventBus.getDefault().register(this)
    }

    @Subscribe(threadMode = ThreadMode.BACKGROUND)
    fun onDeleteNotebookCommand(deleteNotebookCommand: DeleteNotebookCommand) {
        execute(Runnable {
            execute(Runnable { smartNotebookRepository.deleteSmartNotebook(deleteNotebookCommand.smartNotebook!!) }
            )
        })
    }

    @Subscribe(threadMode = ThreadMode.BACKGROUND)
    fun onDeleteNoteCommand(deleteNoteCommand: DeleteNoteCommand) {
        execute(Runnable {
            execute(Runnable {
                smartNotebookRepository.deleteNoteFromBook(
                    deleteNoteCommand.smartNotebook!!,
                    deleteNoteCommand.atomicNote!!
                )
            }
            )
        })
    }
}
